pmlc/dialect/pxa/transforms/stencil.cc

```diff
@@ -270,19 +509,46 @@ void Stencil::DoStenciling() {
   CollectUsedIndices();
   CollectStrideOneIndices();
 
+  auto ranges = op.getConstantRanges();
+  if (!ranges)
+    return;
+
+  opConstRanges = *ranges;
+
+  // Search tensors' order, inner index and their tiles
+  SearchTensorsOrder();
+
+  IVLOG(1, "Best Perf: " << bestPerf);
+  IVLOG(1, "Best Tiles: " << bestTiles[0] << ":" << bestTiles[1] << ":"
+                          << bestTiles[2]);
+
+  if (bestPerf == std::numeric_limits<double>::max()) {
+    IVLOG(1, "No tile plan for stencil.");
+    return;
+  }
+
   op.setAttr("is_gemm", mlir::UnitAttr::get(op.getContext()));
 }
 
 void StencilPass::runOnFunction() {
   auto func = getFunction();
   func.walk([&](mlir::AffineParallelOp op) {
-    Stencil as(op);
+    unsigned threads = numThreads.getValue();
```

Comment:
Let's do this in the constructor of the StencilPass, it should be constructed with a value (never empty).
This isn't an `Optional` (which is possible to hold no value), but an `Option` (which I argue should always hold a value, and I'd be surprised if it can even hold a nothing value).

Reply:
I think the idea here is that '0' threads means: please pick a number of threads for me, which is reasonable enough IMO.  Although, why don't we move this code outside of the loop?

Reply:
I think we should pass `numThreads.getValue()` to the ctor of `Stencil` directly. If someone wants to use hardware_concurrency, they can do it at usage time.

Reply:
Moved.